src/backends/native/meta-renderer-native.c

```diff
@@ -532,32 +533,46 @@ flip_egl_stream (MetaRendererNative *renderer_native,
   CoglDisplay *cogl_display = cogl_context->display;
   CoglRendererEGL *egl_renderer = cogl_display->renderer->winsys;
   EGLAttrib *acquire_attribs;
+  EGLint egl_error;
   GError *error = NULL;
 
+  *fb_in_use = FALSE;
+
   if (renderer_native->egl.no_egl_output_drm_flip_event)
-    return;
+    return FALSE;
 
   acquire_attribs = (EGLAttrib[]) {
     EGL_DRM_FLIP_EVENT_DATA_NV,
     (EGLAttrib) flip_closure,
     EGL_NONE
   };
-  if (!meta_egl_stream_consumer_acquire_attrib (egl,
-                                                egl_renderer->edpy,
-                                                onscreen_native->egl.stream,
-                                                acquire_attribs,
-                                                &error))
+
+  egl_error = meta_egl_stream_consumer_acquire_attrib (egl,
+                                                       egl_renderer->edpy,
+                                                       onscreen_native->egl.stream,
+                                                       acquire_attribs,
+                                                       &error);
+  if (egl_error != EGL_SUCCESS)
     {
-      g_warning ("Failed to flip EGL stream (%s), relying on clock from now on",
-                 error->message);
+      if (egl_error == EGL_RESOURCE_BUSY_EXT)
+        {
+          g_warning ("Failed to flip EGL stream (%s)", error->message);
+          *fb_in_use = TRUE;
```

Comment:
This should probably not set to TRUE, since the call didn't have any effect. Then again, "fb_in_use" is not a thing in EGLDevice paths (as its ignored) but it seems I had it set always anyway. I also think the warning is not needed. It's expected to happen, so it shouldn't go into logs as warnings.